karabo_data/reader.py

```diff
@@ -1058,6 +1062,52 @@ def RunDirectory(path):
 RunHandler = RunDirectory
 
 
+def open_run(proposal, run, proc=True):
+    """Access data from a specified run in the EuXFEL files on Maxwell.
+
+    Parameters
+    ----------
+    proposal: str, int
+        A proposal number, such as 2012, '2012', 'p002012', or a path such as
+        '/gpfs/exfel/exp/SPB/201701/p002012'.
+    run: str, int
+        A run number such as 243, '243' or 'r0243'.
+    proc: bool
+        If True (default), combine data from the raw and 'proc' (processed)
+        folders for the specified run. If False, access only the raw data.
+    """
+    if isinstance(proposal, int):
+        proposal = 'p{:06d}'.format(proposal)
+    elif ('/' not in proposal) and not proposal.startswith('p'):
+        proposal = 'p' + proposal.rjust(6, '0')
+
+    prop_dir = find_proposal(proposal)
+
+    if isinstance(run, int):
+        run = 'r{:04d}'.format(run)
+    elif not run.startswith('r'):
+        run = 'r' + run.rjust(4, '0')
+
+    if not proc:
+        return RunDirectory(osp.join(prop_dir, 'raw', run))
+
+    raw_files = glob(osp.join(prop_dir, 'raw', run, '*.h5'))
+    proc_files = glob(osp.join(prop_dir, 'proc', run, '*.h5'))
+
+    # Names look like RAW-R0243-AGIPD10-S00002.h5 . If the 3rd part ('AGIPD10')
+    # exists in proc, skip loading it from raw.
+    got_in_proc = {osp.basename(p).split('-')[2] for p in proc_files}
+    raw_files = [p for p in raw_files
+                 if osp.basename(p).split('-')[2] not in got_in_proc]
+
+    files = proc_files + raw_files
+    if not files:
+        raise Exception("No .h5 files found for proposal {!r} run {!r}"
+                        .format(proposal, run))
```

Comment:
I wonder if it would be more helpful here to show the resolved path

Reply:
Agreed